File: model/src/iris/Iris.java
package iris;

import data.ClassifiedData;

public class Iris implements ClassifiedData {

    enum IrisClass {
        SETOSA,
        VERSICOLOUR,
        VIRGINICA
    }

    private final double sepalLength;
    private final double sepalWidth;
    private final double petalLength;
    private final double petalWidth;
    private final IrisClass irisClass;

    private final double[] vector;

    public Iris(double sepalLength, double sepalWidth, double petalLength, double petalWidth, IrisClass irisClass) {
        this.sepalLength = sepalLength;
        this.sepalWidth = sepalWidth;
        this.petalLength = petalLength;
        this.petalWidth = petalWidth;
        this.irisClass = irisClass;

        vector = new double[]{
                sepalLength,
                sepalWidth,
                petalLength,
                petalWidth
        };
    }

    @Override
    public double[] asVector() {
        return vector;
    }

    @Override
    public int getClassId() {
        switch (irisClass) {
            case SETOSA:
                return 0;
            case VERSICOLOUR:
                return 1;
            case VIRGINICA:
                return 2;
            default:
                throw new RuntimeException("Unexpected class");
        }
    }
}

File: model/src/iris/Iris.kt
package iris

import data.DataWithResult

class Iris(sepalLength: Double,
           sepalWidth: Double,
           petalLength: Double,
           petalWidth: Double,
           private val irisClass: Iris.IrisClass) : DataWithResult(doubleArrayOf(sepalLength, sepalWidth, petalLength, petalWidth), 0.0) {

    enum class IrisClass {
        SETOSA,
        VERSICOLOUR,
        VIRGINICA
    }

    override val result: Double
        get() {
            return when (irisClass) {
                Iris.IrisClass.SETOSA -> 0.0
                Iris.IrisClass.VERSICOLOUR -> 1.0
                Iris.IrisClass.VIRGINICA -> 2.0
            }
        }
}
